content/en/docs/ops/configuration/traffic-management/network-topologies/index.md

```diff
@@ -0,0 +1,156 @@
+of the Istio gateway proxy so that the client address can be extracted correctly.
+
+To set the number of trusted proxies add the following to your `topology.yaml` file.
+
+{{< text yaml >}}
+apiVersion: install.istio.io/v1alpha1
+kind: IstioOperator
+spec:
+  meshConfig:
+    defaultConfig:
+      gatewayTopology:
+        numTrustedProxies: <VALUE>
+{{< /text >}}
+
+For example, if you have a cloud based LB, a reverse proxy and then the Istio gateway proxy
+then `<VALUE>` would be 2.
+
+{{< warning >}}
+Note that all the proxies in front need of Istio gateway proxy must parse HTTP traffic and append the X-Forwarded-For
+headers at each hop. If the number of entries in X-Forwarded-For header is less than the number of
+trusted hops configured, Envoy falls back to using the immediate downstream address as the trusted
+client address. Please refer to [Envoy documentation](https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_conn_man/headers#x-forwarded-for)
+to understand how X-Forwarded-For headers and trusted client addresses are determined.
+{{< /warning >}}
+
+See [Envoy documentation](https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_conn_man/headers#x-forwarded-for)
+for more information about this capability.
+
+#### Example using XFF capability with httpbin
+
+1. Specify `numTrustedProxies` as 2 either through `MeshConfig` or an `proxy.istop/io/config` annotation.
+2. Create `httpbin` namespace
+    {{< text bash >}}
+    $ kubectl create namespace httpbin
+    namespace/httpbin created
+    {{< /text >}}
+3. Set the `istio-injection` label to `enabled` for sidecar injection
+    {{< text bash >}}
+    $ kubectl label --overwrite namespace httpbin istio-injection=enabled
+    namespace/httpbin labeled
+    {{< /text >}}
+4. Deploy `httpbin` in the `httpbin` namespace
+    {{< text bash >}}
+    $ kubectl apply -n httpbin -f samples/httpbin/httpbin.yaml
+    {{< /text >}}
+5. Deploy a gateway associated with `httpbin`
+    {{< text bash >}}
+    $ kubectl apply -n httpbin -f samples/httpbin/httpbin-gateway.yaml
+    {{< /text >}}
+6. Set a local `$GATEWAY_URL` environmental variable based on your Istio ingress gateway's IP address
+    {{< text bash >}}
+    $ export GATEWAY_URL=$(kubectl -n istio-system get service istio-ingressgateway -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
+    {{< /text >}}
+7. Run the following `curl` command to verify the `X-Envoy-External-Address` and `X-Forwarded-For` are set correctly.
+    {{< text bash >}}
+    $ curl -H 'X-Forwarded-For: 56.5.6.7, 72.9.5.6, 98.1.2.3' $GATEWAY_URL/get?show_env=true
+    {
+      "args": {
+        "show_env": "true"
+      },
+      "headers": {
+        ...
+        "X-Envoy-External-Address": "72.9.5.6",
+        ...
+        "X-Forwarded-For": "56.5.6.7, 72.9.5.6, 98.1.2.3, <YOUR GATEWAY IP>",
+        ...
+      },
+      ...
+    }    
+    {{< /text >}}
+
```

Comment:
Note that the `X-Envoy-External-Address` is set to the "second" from last address in `X-Forwarded-For` header as per your `numTrustedProxies` setting. Additionally, the gateway workload appends its IP in the `X-Forwarded-For` header before forwarding it to the upstream `httpbin` workload.

Reply:
Added